libs/live-nft-react/src/hooks/useSpamTxFiltering.ts

```diff
@@ -0,0 +1,133 @@
+import { getAccountCurrency } from "@ledgerhq/coin-framework/account/helpers";
+import { BlockchainsType, BlockchainEVM } from "@ledgerhq/live-nft/supported";
+import { NFTResource } from "@ledgerhq/live-nft/types";
+import {
+  Operation,
+  AccountLike,
+  DailyOperations,
+  NFTCollectionMetadataResponse,
+} from "@ledgerhq/types-live";
+import { useMemo } from "react";
+import { useNftCollectionMetadataBatch } from "../NftMetadataProvider";
+
+type AccountMap = Record<string, AccountLike>;
+
+type NftOp = {
+  contract: string;
+  currencyId: string;
+};
+
+type NftOperations = Record<string, NftOp>;
+
+export function useNftOperations({
+  accountsMap,
+  groupedOperations,
+  spamFilteringTxEnabled,
+}: {
+  accountsMap: AccountMap;
+  groupedOperations?: DailyOperations;
+  spamFilteringTxEnabled: boolean;
+}) {
+  return useMemo(
+    () =>
+      spamFilteringTxEnabled
+        ? groupedOperations?.sections
+            .flatMap(section => section.data)
+            .reduce((acc, operation) => {
+              if (operation.type === "NFT_IN") {
+                const account = accountsMap[operation.accountId];
+                const contract = operation.contract || "";
+                acc[contract] = {
+                  contract,
+                  currencyId: getAccountCurrency(account).id,
+                };
+              }
+              return acc;
+            }, {} as NftOperations) || ({} as NftOperations)
+        : ({} as NftOperations),
+    [spamFilteringTxEnabled, groupedOperations, accountsMap],
+  );
+}
+
+export function filterSection(
+  operations: Operation[],
+  metadataMap: Map<string | undefined, number>,
+  accountsMap: AccountMap,
+  markNftAsSpam?: (collectionId: string, blockchain: BlockchainsType, spamScore: number) => void,
+  threshold?: number,
+) {
+  const filtered: Operation[] = [];
```

Comment:
Why not just using a Map from the start:
```suggestion
  const filtered = new Map<string, Operation>();
```
and instead of `filtered.push(item)`: `filtered.set(item.id, item)`

Reply:
Indeed, I just saw that while reading code x)